lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -238,5 +252,289 @@ bool circt::firrtl::fromJSON(json::Value &value,
     annotationMap[a] = ArrayAttr::get(context, mutableAnnotationMap[a]);
   }
 
+  scatterCustomAnnotations(annotationMap, context, annotationID);
+
   return true;
 }
+
+enum class CustomAnnotation {
+  Unknown,
+  NotImplemented,
+  DataTap,
+  MemTap,
+  SerializedView
+};
+
+/// Recursively walk a sifive.enterprise.grandcentral.AugmentedType to extract
+/// any annotations it may contain.  This is going to generate two types of
+/// annotations:
+///   1) Annotations necessary to build interfaces and store them at "~"
+///   2) Scattered annotations for how components bind to interfaces
+static void parseAugmentedType(
+    MLIRContext *context, DictionaryAttr augmentedType,
+    llvm::StringMap<llvm::SmallVector<Attribute>> &newAnnotations,
+    const StringRef &companion, StringAttr name, StringAttr defName) {
```

Comment:
StringRef is just a pointer+length, so please pass it by value instead of by const&

Reply:
Will be fixed in the rebase.